ext/standard/file.c

```diff
@@ -543,7 +543,7 @@ PHP_FUNCTION(file_get_contents)
 		RETURN_FALSE;
 	}
 
-	if (offset > 0 && php_stream_seek(stream, offset, SEEK_SET) < 0) {
+	if (offset != 0 && php_stream_seek(stream, offset, ((offset > 0) ? SEEK_SET : SEEK_END)) < 0) {
```

Comment:
Also thought about it, but did not find a way to combine such feature with filters modifying input lengths. The only reliable solution would be to read everything in memory and, then, take a substring but, in most cases, we don't want to read more than needed. Reading the whole stream also implies that the stream has an end, which is not always the case. So, defining a byte count from the stream end is not an easy concept.